message_sender/serializers.py

```diff
@@ -1,34 +1,20 @@
 from .models import Inbound, Outbound, OutboundSendFailure, Channel
 from rest_hooks.models import Hook
 from rest_framework import serializers
-from rest_framework.exceptions import APIException
-
-
-class IncorrectChannel(APIException):
-    status_code = 400
-    default_detail = 'The channel supplied is not configured.'
-    default_code = 'incorrect_channels'
 
 
 class OutboundSerializer(serializers.HyperlinkedModelSerializer):
 
+    channel = serializers.PrimaryKeyRelatedField(
+        queryset=Channel.objects.all(), required=False)
```

Comment:
It does this `return self.get_queryset().get(pk=data)` inside `to_internal_value` so I'm guessing it gets evaluated on run time.